internal/uvm/scsi.go

```diff
@@ -66,32 +69,54 @@ func (uvm *UtilityVM) findSCSIAttachment(findThisHostPath string) (int, int32, s
 //
 // hostPath is required
 // uvmPath is optional.
+// isLayer indicates that this is a read-only (LCOW) layer VHD too big to fit on vpmem.
+// calculateUVMPath indicates that the UVM path is calculated based on /tmp/SC/L
 //
 // Returns the controller ID (0..3) and LUN (0..63) where the disk is attached.
-func (uvm *UtilityVM) AddSCSI(hostPath string, uvmPath string) (int, int32, error) {
+func (uvm *UtilityVM) AddSCSI(hostPath string, uvmPath string, isLayer bool, calculateUVMPath bool) (int, int32, error) {
```

Comment:
Part of me thinks that you should make
```go
func (uvm *UtilityVM) AddSCSILayer(hostPath string) (int, int32, error) {
    //...
}
func (uvm *UtilityVM) AddSCSI(hostPath string, uvmPath) (int, int32, error) {
    //...
}
```
This overload makes it very easy to understand what it does and takes care of the `isLayer` bool, and the `calculateUVMPath` because its implied based on it being a Layer. It also means that it should only be called on LCOW which is an easy precondition. 

I don't care if you unify them behind a private single function but should we keep both public ones?

Reply:
Sure, easy to change. Now or in a followup?

Reply:
If you dont mind on this one can we do it now. I think its much easier to parse